From 7d7e669c606e72ce8c20f19a0eacacd47522ca1a Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Sat, 28 Sep 2024 11:32:49 -0700
Subject: [PATCH] lib/Kconfig.debug: Force disable BUILTIN_MODULE_RANGES

Commit 5f5e7344322f ("kbuild: generate offset range data for builtin
modules") introduced a dependency on gawk, which causes allmodconfig
builds in TuxSuite to fail because gawk is not present:

  /bin/sh: 1: gawk: not found
  make[3]: *** [/builds/linux/scripts/Makefile.vmlinux:47: modules.builtin.ranges] Error 127

While this is fixed in TuxSuite, force disable
CONFIG_BUILTIN_MODULE_RANGES, as it is not critical to test.

Link: https://gitlab.com/Linaro/tuxmake/-/issues/220
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 lib/Kconfig.debug | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 7315f643817a..0a1fbf9ebfd6 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -575,6 +575,7 @@ config VMLINUX_MAP
 
 config BUILTIN_MODULE_RANGES
 	bool "Generate address range information for builtin modules"
+	depends on BROKEN
 	depends on !LTO
 	depends on VMLINUX_MAP
 	help
-- 
2.46.1

